payjoin/tests/integration.rs

```diff
@@ -400,6 +409,12 @@ mod integration {
                 assert!(response.status().is_success());
```

Comment:
you should be able to check that this `[encapsulated_]response`'s processed inner http response is_none here rather than making another request.